### PR TITLE
[Growth 38][16] As a user, after choosing AWS infrastructure variant, I can choose the AWS region

### LINKED ISSUE #16: [Growth 38] As a user, after choosing AWS infrastructure variant, I can choose the AWS region.
## Why
Once choosing an AWS infrastructure variant, the app should allow the user to specify the region.

## Acceptance Criteria
After the user choose an infrastructure variant, the app asks the user to specify the region through the CLI

### PR DESCRIPTION
**Pull request by byhbt**
Closes #16 

## What happened 👀

- Implement prompt user for the AWS Region

## Insight 📝

n/a

## Proof Of Work 📹

https://user-images.githubusercontent.com/948856/170160575-dc589dcc-154f-4fd0-84b1-027b09d6701f.mp4

### CODE REVIEW

**Comment by longnd** [comment]
@byhbt should we return a message if the user chooses a non-supported cloud provider, e.g. GCP? sth like:
> This cloud provider hasn't been supported.

**Comment by byhbt** [comment]
> @byhbt should we return a message if the user chooses a non-supported cloud provider, e.g. GCP? sth like:
> 
> > This cloud provider hasn't been supported.

Yup i will update it today.

**Comment by byhbt** [comment]
> @byhbt should we return a message if the user chooses a non-supported cloud provider, e.g. GCP? sth like:
> 
> > This cloud provider hasn't been supported.

Screen Shot 2022-05-25 at 17 05 25

Added @longnd

**Review by longnd — COMMENTED** [review]
@byhbt could you do a favor by putting the region in the variables.tf file. Currently the variable is hard-coded
https://github.com/nimblehq/infrastructure-templates/blob/1e1f55626920eadabcd05844a101dcd18ec56804/skeleton/aws/variables.tf#L11-L14

**Comment by andyduong1920** [comment]
Not sure why do we have some changes in the `yarn.lock` file on this PR @byhbt

File: src/commands/generate/index.ts [hunk 1]
```diff
@@ -72,14 +73,20 @@ export default class Hello extends Command {
             },
           ],
         },
+        {
+          type: 'input',
+          name: 'awsRegion',
+          message: 'Which AWS Region do you choose? ',
```
Comment (hoangmirs) [inline src/commands/generate/index.ts:79]: Should we add the default value `ap-southeast-1` as we often use that region?

**Comment by byhbt** [comment]
> 1b3bcfa

Added in : https://github.com/nimblehq/infrastructure-templates/pull/35/commits/1b3bcfacdbac79d0971930c20a8d35d4f2cfa96c

File: src/commands/generate/index.ts (hunk 1 above)
Comment (byhbt) [reply src/commands/generate/index.ts]: Added default region here: https://github.com/nimblehq/infrastructure-templates/pull/35/commits/20c518befd7c03b64fefb53bf438fe79bebd8374

**Comment by byhbt** [comment]
> @byhbt could you do a favor by putting the region in the variables.tf file. Currently the variable is hard-coded
> 
> https://github.com/nimblehq/infrastructure-templates/blob/1e1f55626920eadabcd05844a101dcd18ec56804/skeleton/aws/variables.tf#L11-L14

Added here: https://github.com/nimblehq/infrastructure-templates/pull/35/commits/20c518befd7c03b64fefb53bf438fe79bebd8374

File: src/templates/aws/advanced.ts [hunk 2]
```diff
@@ -34,6 +35,17 @@ export default class Advanced {
...
   }
 
+  private applyRegion(): void {
+    const regionVariableContent = dedent`
+    \nvariable "region" {
+      description = "AWS region"
+      type        = string
+      default     = "${this.options.awsRegion}"
+    }
+    \n`
```
Comment (longnd) [inline src/templates/aws/advanced.ts:45]: we're using `\n\n` in the bellow methods, should we stick to the same way?
if you prefer to wrap the content inside `\n{content}\n`, then we should change the following methods for consistency.

Comment (byhbt) [reply src/templates/aws/advanced.ts]: yup, in order to keep the `region` variable consistent with other variable, i update the `\n` like others.
And add one more space in the `variables.tf` in the `skeleton` here: https://github.com/nimblehq/infrastructure-templates/pull/35/commits/6a544745a73b7d5b2691eda347abfe317d459853

**Review by longnd — APPROVED** [review]

**Review by junan — APPROVED** [review]

**Pull request merged**